### PR TITLE
New hostgroup features

### PR DESCRIPTION
**Pull request by sidkhemka**
This pull request adds two new features to the hostgroups view
1. adds a hosts tab on the edit page of a hostgroup to view a list of hosts belonging to the hostgroup, and easily access their links
   screen shot 2014-08-21 at 3 37 52 pm
2. adds a new display action - this is linked to on the hostgroup index page for each hostgroup IF the user does not have edit permissions. This essentially renders the same page as the edit action would, but runs a Javascript function to disable all edit fields and remove buttons so the user can only view details, not edit them, as a user with only view hostgroup permissions could only see a list of hostgroups earlier, not the details of each individual hostgroup. 
   screen shot 2014-09-05 at 4 51 15 pm

### CODE REVIEW

**Comment by dLobatog** [comment]
[test] <- this triggers a test run of your PR on our Jenkins CI system

Thanks @sidkhemka for your PR. I tested it a little bit and I found a few issues:
- The index page for hostgroups suddenly looks like this:
  
- Hosts tab only shows hosts directly inherited. I think it would be useful to also show other hosts that can be inheriting properties from the selected Host group.
- Hosts tab should be paginated with AJAX if possible.

File: app/views/hostgroups/_grouphosts.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <th class="hidden-xs"><%= sort :os, :as => _("Operating system") %></th>
+    <th class="hidden-xs"><%= sort :environment, :as => _('Environment') %></th>
+    <th class="hidden-tablet hidden-xs"><%= sort :model, :as => _('Model') %></th>
+    <th class="hidden-tablet hidden-xs"><%= sort :last_report, :as => _('Last report') %></th>
+	</tr>
+	<% if hosts.any? %>
+		<% hosts.each do |host| %>
+			<tr>
+      			<td class='ellipsis'><%= name_column(host) %> </td>
+      			<td class="hidden-xs"><%= (icon(host.os, :size => "18x18") + trunc(" #{host.os.to_label}",14)).html_safe if host.os %></td>
```
Comment (dLobatog) [inline app/views/hostgroups/_grouphosts.html.erb]: Please if `host.os.present?` when checking if a relation object is present.

**Comment by sidkhemka** [comment]
Thanks @eLobato 
I've made the fixes for all 4 (hostgroup index page buttons, pagination on hosts tab, displaying hosts from child hostgroups and using .present?)

EDIT: still needs ajax support for pagination, working on that now

screen shot 2014-09-08 at 2 27 32 pm

screen shot 2014-09-08 at 2 46 05 pm

**Comment by sidkhemka** [comment]
Updated with the edits including ajax pagination

**Comment by orrabin** [comment]
[test]

**Comment by shlomizadok** [comment]
@sidkhemka can you please rebase?

**Comment by dLobatog** [comment]
[test] 

This would be a good nicety in my view, however some people might complain about loading times, and also maintenance of this wouldn't be trivial. I'd say this could very well be a plugin, @ohadlevy @domcleal @lzap any thoughts..?

**Comment by lzap** [comment]
I don't expect one additional db query to be a bottleneck on this page. This one is trivial, we already do bunch of permission queries. I am fine with this to be in core.

**Comment by boaz0** [comment]
[test]

File: app/views/hostgroups/_grouphosts.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <th class="hidden-xs"><%= sort :os, :as => _("Operating system") %></th>
+    <th class="hidden-xs"><%= sort :environment, :as => _('Environment') %></th>
+    <th class="hidden-tablet hidden-xs"><%= sort :model, :as => _('Model') %></th>
+    <th class="hidden-tablet hidden-xs"><%= sort :hostgroup, :as => _("Host group") %></th>
+    <th class="hidden-tablet hidden-xs"><%= sort :last_report, :as => _('Last report') %></th>
+</tr>
+<% if @hosts && @hosts.any? %>
+	<div id="hosts">
+	<% @hosts.each do |host| %>
+		<tr>
```
Comment (dLobatog) [inline app/views/hostgroups/_grouphosts.html.erb:14]: We try to keep ERB indentation to 2 spaces

**Comment by dLobatog** [comment]
@sidkhemka Really sorry this has been dead for some time, could you resubmit it rebasing it against the latest develop branch? I think it would also be better to submit the 'display' feature and the 'list of hosts' feature separately. Both of them would need [a ticket number in Redmine](http://projects.theforeman.org/issues/).

In general they're almost ready to go except for nitpicks with the code, if you could also look into making the Hostgroup-> Hosts tab display the Hosts#index page instead of creating a new page Hostgroup#grouphosts that would be great, as there would be two pages with the same purpose to maintain. 

Let me know what you think, if it's too much we can take it over too and resubmit. I just don't want code to completely rot in the list of open Pull Request :smile: 

Thanks again!!

**Comment by tbrisker** [comment]
@sidkhemka,
Thank you very much for your efforts! 
This PR has been open for a very long time with no activity. 
Please indicate if you plan to continue working on it or not; if not, feel free to close it. 
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Comment by tbrisker** [comment]
Dear @sidkhemka ,
It has been 20 days since my last comment with no replies. 
I am assuming this pull request has been abandoned and closing it.
If you wish to continue working on this in the future, feel free to reopen.
On behalf of the entire Foreman team, I wish to thank you for your efforts here and hope to see more contributions from you in the future.

**Pull request closed without merging**